crates/types/src/consensus.rs

```diff
@@ -824,13 +824,10 @@ impl<TYPES: NodeType> Consensus<TYPES> {
         private_key: &<TYPES::SignatureKey as SignatureKey>::PrivateKey,
         epoch: TYPES::Epoch,
     ) -> Option<()> {
-        let consensus = consensus.upgradable_read().await;
-        let txns = consensus.saved_payloads().get(&view)?;
-        let vid =
-            VidDisperse::calculate_vid_disperse(Arc::clone(txns), &membership, view, epoch, None)
-                .await;
+        let txns = Arc::clone(consensus.read().await.saved_payloads().get(&view)?);
+        let vid = VidDisperse::calculate_vid_disperse(txns, &membership, view, epoch, None).await;
         let shares = VidDisperseShare::from_vid_disperse(vid);
-        let mut consensus = ConsensusUpgradableReadLockGuard::upgrade(consensus).await;
+        let mut consensus = consensus.write().await;
```

Comment:
Before, with the upgradable read, this whole thing used to be atomic, in that consensus state could not change in between us reading the transactions and doing this update. Logically it looks like it doesn't matter, nothing that happens to consensus state is going to affect this update we're doing, since we're just populating potential missing state but not making any effectful changes. But I just want to double check this is intentional

Reply:
Hey yeah - I took a look too - it was not immediately apparent. @bfish713 ?

Reply:
Yeah it's completely fine for it to be non atomic like you both expect